src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSourceRequest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// A non-cached HTTP request handled by <see cref="HttpSource"/>.
+    /// </summary>
+    public class HttpSourceRequest
+    {
+        public static readonly TimeSpan DefaultRequestTimeout = TimeSpan.FromSeconds(100);
+
+        public HttpSourceRequest(string uri, ILogger log)
+            : this(
+                uri,
+                () => HttpRequestMessageFactory.Create(HttpMethod.Get, uri, log))
+        {
+        }
+
+        public HttpSourceRequest(Uri uri, ILogger log)
+            : this(
+                uri?.ToString(),
+                () => HttpRequestMessageFactory.Create(HttpMethod.Get, uri, log))
+        {
+        }
+
+        public HttpSourceRequest(Uri uri, Func<HttpRequestMessage> requestFactory)
+            : this(uri?.ToString(), requestFactory)
+        {
+        }
+
+        public HttpSourceRequest(string uri, Func<HttpRequestMessage> requestFactory)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            if (requestFactory == null)
+            {
+                throw new ArgumentNullException(nameof(requestFactory));
+            }
+
+            Uri = uri;
+            RequestFactory = requestFactory;
+            IgnoreNotFounds = false;
```

Comment:
sure, I'm not too particular about these two alternatives -- glad to align with you.